core/src/main/java/org/apache/iceberg/PositionDeletesTable.java

```diff
@@ -0,0 +1,372 @@
+                partitionType,
+                "Partition that position delete row belongs to"),
+            Types.NestedField.required(
+                POSITION_DELETE_TABLE_SPEC_ID,
+                "spec_id",
+                Types.IntegerType.get(),
+                "Spec ID of the file that the position delete row belongs to"),
+            Types.NestedField.required(
+                POSITION_DELETE_TABLE_FILE_PATH,
+                "delete_file_path",
+                Types.StringType.get(),
+                "Spec ID of the file that the position delete row belongs to"));
+
+    if (partitionType.fields().size() > 0) {
+      return result;
+    } else {
+      // avoid returning an empty struct, which is not always supported. instead, drop the partition
+      // field
+      return TypeUtil.selectNot(result, Sets.newHashSet(POSITION_DELETE_TABLE_PARTITION_FIELD_ID));
+    }
+  }
+
+  public static class PositionDeletesTableScan
+      extends SnapshotScan<BatchScan, ScanTask, ScanTaskGroup<ScanTask>> implements BatchScan {
+
+    protected PositionDeletesTableScan(TableOperations ops, Table table, Schema schema) {
+      super(ops, table, schema, new TableScanContext());
+    }
+
+    protected PositionDeletesTableScan(
+        TableOperations ops, Table table, Schema schema, TableScanContext context) {
+      super(ops, table, schema, context);
+    }
+
+    @Override
+    protected PositionDeletesTableScan newRefinedScan(
+        TableOperations newOps, Table newTable, Schema newSchema, TableScanContext newContext) {
+      return new PositionDeletesTableScan(newOps, newTable, newSchema, newContext);
+    }
+
+    @Override
+    public CloseableIterable<ScanTaskGroup<ScanTask>> planTasks() {
+      CloseableIterable<ScanTask> scanTasks = planFiles();
+      return TableScanUtil.planTaskGroups(
+          scanTasks, targetSplitSize(), splitLookback(), splitOpenFileCost());
+    }
+
+    @Override
+    protected CloseableIterable<ScanTask> doPlanFiles() {
+      Expression rowFilter = context().rowFilter();
+      String schemaString = SchemaParser.toJson(tableSchema());
+
+      Map<Integer, PartitionSpec> transformedSpecs =
+          table().specs().entrySet().stream()
+              .map(
+                  e ->
+                      Pair.of(
+                          e.getKey(), BaseMetadataTable.transformSpec(tableSchema(), e.getValue())))
+              .collect(Collectors.toMap(Pair::first, Pair::second));
+
+      CloseableIterable<ManifestFile> deleteManifests =
+          CloseableIterable.withNoopClose(snapshot().deleteManifests(tableOps().io()));
+      CloseableIterable<CloseableIterable<ScanTask>> results =
+          CloseableIterable.transform(
+              deleteManifests,
+              m -> {
+                // Filter partitions
+                CloseableIterable<ManifestEntry<DeleteFile>> deleteFileEntries =
+                    ManifestFiles.readDeleteManifest(m, tableOps().io(), transformedSpecs)
```

Comment:
Do we need to pass a projection while reading delete manifests?

Reply:
I assume you mean, with/without stats?  Fixed.  Took me a little bit to figure it out, I needed to add a DELETE_SCAN_COLUMNS here that has content, versus the base table's SCAN_COLUMNS which does not.  As I have a filter on position_delete file content.